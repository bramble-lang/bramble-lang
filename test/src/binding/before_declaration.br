fn my_main() -> i64 {
    root::std::io::writei64ln(x);
    let x: i64 := 5;
    return;
}