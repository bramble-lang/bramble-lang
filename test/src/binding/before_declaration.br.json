[{"id": 1, "stage": "lexer", "source": [0, 2], "ok": "Keyword"},
{"id": 2, "stage": "lexer", "source": [3, 10], "ok": "Identifier"},
{"id": 3, "stage": "lexer", "source": [10, 11], "ok": "Operator"},
{"id": 4, "stage": "lexer", "source": [11, 12], "ok": "Operator"},
{"id": 5, "stage": "lexer", "source": [13, 15], "ok": "Operator"},
{"id": 6, "stage": "lexer", "source": [16, 19], "ok": "Primitive"},
{"id": 7, "stage": "lexer", "source": [20, 21], "ok": "Operator"},
{"id": 8, "stage": "lexer", "source": [26, 33], "ok": "Keyword"},
{"id": 9, "stage": "lexer", "source": [33, 35], "ok": "Operator"},
{"id": 10, "stage": "lexer", "source": [35, 38], "ok": "Identifier"},
{"id": 11, "stage": "lexer", "source": [38, 40], "ok": "Operator"},
{"id": 12, "stage": "lexer", "source": [40, 42], "ok": "Identifier"},
{"id": 13, "stage": "lexer", "source": [42, 44], "ok": "Operator"},
{"id": 14, "stage": "lexer", "source": [44, 54], "ok": "Identifier"},
{"id": 15, "stage": "lexer", "source": [54, 55], "ok": "Operator"},
{"id": 16, "stage": "lexer", "source": [55, 56], "ok": "Identifier"},
{"id": 17, "stage": "lexer", "source": [56, 57], "ok": "Operator"},
{"id": 18, "stage": "lexer", "source": [57, 58], "ok": "Operator"},
{"id": 19, "stage": "lexer", "source": [63, 66], "ok": "Keyword"},
{"id": 20, "stage": "lexer", "source": [67, 68], "ok": "Identifier"},
{"id": 21, "stage": "lexer", "source": [68, 69], "ok": "Operator"},
{"id": 22, "stage": "lexer", "source": [70, 73], "ok": "Primitive"},
{"id": 23, "stage": "lexer", "source": [74, 76], "ok": "Operator"},
{"id": 24, "stage": "lexer", "source": [77, 78], "ok": "Number"},
{"id": 25, "stage": "lexer", "source": [78, 79], "ok": "Operator"},
{"id": 26, "stage": "lexer", "source": [84, 90], "ok": "Keyword"},
{"id": 27, "stage": "lexer", "source": [91, 92], "ok": "Number"},
{"id": 28, "stage": "lexer", "source": [92, 93], "ok": "Operator"},
{"id": 29, "stage": "lexer", "source": [94, 95], "ok": "Operator"},
{"id": 35, "parent_id": 33, "stage": "parser", "source": [16, 19], "ok": "Primitive Type"},
{"id": 33, "parent_id": 32, "stage": "parser", "source": [0, 19], "ok": "Routine Declaration"},
{"id": 49, "parent_id": 48, "stage": "parser", "source": [26, 54], "ok": "Path"},
{"id": 60, "parent_id": 59, "stage": "parser", "source": [55, 56], "ok": "Path"},
{"id": 59, "parent_id": 57, "stage": "parser", "source": [55, 56], "ok": "Identifier"},
{"id": 57, "parent_id": 55, "stage": "parser", "source": [55, 56]},
{"id": 55, "parent_id": 54, "stage": "parser", "source": [55, 56]},
{"id": 54, "parent_id": 53, "stage": "parser", "source": [55, 56]},
{"id": 53, "parent_id": 52, "stage": "parser", "source": [55, 56]},
{"id": 52, "parent_id": 51, "stage": "parser", "source": [55, 56]},
{"id": 51, "parent_id": 50, "stage": "parser", "source": [55, 56]},
{"id": 50, "parent_id": 48, "stage": "parser", "source": [55, 56]},
{"id": 48, "parent_id": 46, "stage": "parser", "source": [26, 57], "ok": "Routine Call"},
{"id": 46, "parent_id": 44, "stage": "parser", "source": [26, 57]},
{"id": 44, "parent_id": 43, "stage": "parser", "source": [26, 57]},
{"id": 43, "parent_id": 42, "stage": "parser", "source": [26, 57]},
{"id": 42, "parent_id": 41, "stage": "parser", "source": [26, 57]},
{"id": 41, "parent_id": 40, "stage": "parser", "source": [26, 57]},
{"id": 40, "parent_id": 39, "stage": "parser", "source": [26, 57]},
{"id": 39, "parent_id": 36, "stage": "parser", "source": [26, 57]},
{"id": 36, "parent_id": 32, "stage": "parser", "source": [26, 58], "ok": "Statement Expression"},
{"id": 65, "parent_id": 64, "stage": "parser", "source": [70, 73], "ok": "Primitive Type"},
{"id": 64, "parent_id": 63, "stage": "parser", "source": [67, 73], "ok": "Identifier Declaration"},
{"id": 78, "parent_id": 74, "stage": "parser", "source": [77, 78], "ok": "Number"},
{"id": 74, "parent_id": 72, "stage": "parser", "source": [77, 78]},
{"id": 72, "parent_id": 71, "stage": "parser", "source": [77, 78]},
{"id": 71, "parent_id": 70, "stage": "parser", "source": [77, 78]},
{"id": 70, "parent_id": 69, "stage": "parser", "source": [77, 78]},
{"id": 69, "parent_id": 68, "stage": "parser", "source": [77, 78]},
{"id": 68, "parent_id": 67, "stage": "parser", "source": [77, 78]},
{"id": 67, "parent_id": 63, "stage": "parser", "source": [77, 78]},
{"id": 63, "parent_id": 62, "stage": "parser", "source": [63, 78], "ok": "Let Binding"},
{"id": 62, "parent_id": 32, "stage": "parser", "source": [63, 79], "ok": "Statement Bind"},
{"id": 109, "parent_id": 105, "stage": "parser", "source": [91, 92], "ok": "Number"},
{"id": 105, "parent_id": 103, "stage": "parser", "source": [91, 92]},
{"id": 103, "parent_id": 102, "stage": "parser", "source": [91, 92]},
{"id": 102, "parent_id": 101, "stage": "parser", "source": [91, 92]},
{"id": 101, "parent_id": 100, "stage": "parser", "source": [91, 92]},
{"id": 100, "parent_id": 99, "stage": "parser", "source": [91, 92]},
{"id": 99, "parent_id": 98, "stage": "parser", "source": [91, 92]},
{"id": 98, "parent_id": 97, "stage": "parser", "source": [91, 92]},
{"id": 97, "parent_id": 32, "stage": "parser", "source": [84, 92], "ok": "Return"},
{"id": 32, "parent_id": 30, "stage": "parser", "source": [0, 95], "ok": "Funtion Definition"},
{"id": 30, "stage": "parser", "source": [0, 95], "ok": "File Module"},
{"id": 112, "stage": "canonize-item-path", "source": [0, 95], "ok": "$before_declaration::before_declaration"},
{"id": 113, "stage": "canonize-item-path", "source": [0, 95], "ok": "$before_declaration::before_declaration::before_declaration"},
{"id": 114, "stage": "canonize-item-path", "source": [0, 95], "ok": "$before_declaration::before_declaration::my_main"},
{"id": 117, "parent_id": 116, "stage": "type-resolver", "source": [55, 56], "error": "L2: Could not find definition for x in this scope"},
{"id": 116, "parent_id": 115, "stage": "type-resolver", "source": [55, 56], "error": "L2: Could not find definition for x in this scope"},
{"id": 115, "stage": "type-resolver", "source": [55, 56], "error": "L2: Could not find definition for x in this scope"}]