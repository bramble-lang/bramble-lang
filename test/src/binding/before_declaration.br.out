Error: L2: x is not defined
